### PR TITLE
Oracle Python Upload Script

### PR DESCRIPTION
**Pull request by rshade**

### CODE REVIEW

File: oracle/OracleBillDownload.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+reporting_namespace = 'bling'
+
+# Download all usage and cost files. You can comment out based on the specific need:
+prefix_file = ""                     #  For cost and usage files
+# prefix_file = "reports/cost-csv"   #  For cost
+# prefix_file = "reports/usage-csv"  #  For usage
+
+# Update these values
+destintation_path = 'CBI'
```
Comment (FabriceScoupeFlexera) [inline oracle/OracleBillDownload.py:26]: Typo: `destination_path`

File: oracle/OracleBillDownload.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import oci
+import os
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+import pytz
+import json
+
+# This script downloads all of the cost, usage, (or both) reports for a tenancy (specified in the config file).
```
Comment (FabriceScoupeFlexera) [inline oracle/OracleBillDownload.py:8]: Very minor: in python, if you move those comments at the beginning of your script as a multi-line string, eg:
```
'''
This script downloads all of the cost [...]

Pre-requisites: [...]
'''
import oci
[...]
```

it is automatically put into the `__doc__` built-in variable. See https://github.com/flexera/optima-bill/blob/master/examples/python3/bill_upload_example.py for an example.

File: oracle/OracleBillDownload.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+# Download all usage and cost files. You can comment out based on the specific need:
+prefix_file = ""                     #  For cost and usage files
+# prefix_file = "reports/cost-csv"   #  For cost
+# prefix_file = "reports/usage-csv"  #  For usage
+
+# Update these values
+destintation_path = 'CBI'
+
+# Download all file or just past three months.
+env_download_all = os.environ.get("DOWNLOAD_ALL")
```
Comment (FabriceScoupeFlexera) [inline oracle/OracleBillDownload.py:29]: Which values can the environment variable `DOWNLOAD_ALL` have?
You can set a default value, used if `DOWNLOAD_ALL` is not defined in the environment, e.g.:
`env_download_all = os.environ.get("DOWNLOAD_ALL", False)`

File: oracle/OracleBillDownload.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+download_all_files = bool(env_download_all)
```
Comment (FabriceScoupeFlexera) [inline oracle/OracleBillDownload.py:30]: `bool(s)` returns `True` if the string `s` is not empty, else `False`. So `bool("False")` returns `True`, and `bool("")` returns False.

File: oracle/OracleBillDownload.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+# Make a directory to receive reports
+if not os.path.exists(destintation_path):
```
Comment (FabriceScoupeFlexera) [inline oracle/OracleBillDownload.py:33]: Same typo here: `destination_path`

File: oracle/OracleBillDownload.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    os.mkdir(destintation_path)
```
Comment (FabriceScoupeFlexera) [inline oracle/OracleBillDownload.py:34]: and here, `destination_path`

File: oracle/OracleBillDownload.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  elif o.time_modified >= last_three_months:
+    print("Downloading files from " + last_three_months.strftime("%Y-%m") + " to " + now.strftime("%Y-%m"))
+    download = True
+  else:
+    download = False
+
+  if download:
+    print('Found file ' + o.name)
+    folder_time = o.time_modified.strftime("%Y-%m")
+    download_folder = os.path.join(destintation_path,folder_time)
```
Comment (FabriceScoupeFlexera) [inline oracle/OracleBillDownload.py:62]: Typo: `destination_path`

File: oracle/OracleBillDownload.py [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    # Make a directory to receive reports
+    if not os.path.exists(download_folder):
+      os.mkdir(download_folder)
+
+    filename = o.name.rsplit('/', 1)[-1]
```
Comment (FabriceScoupeFlexera) [inline oracle/OracleBillDownload.py:67]: Not portable: use this instead: `filename = os.path.basename(o.name)`

**Review by FabriceScoupeFlexera — COMMENTED** [review]
Just a few comments, but nothing major

**Review by gonzalez — APPROVED** [review]

**Pull request merged**